File: Heaps/Sort-K-Sorted-Array/sort.py
import heapq

class Solution:
    def nearlySorted(self, arr, k):  
        #code here
        n = len(arr);
        ans = [0] * n
        heap = []
        j=0
        for i in range(n):
            if len(heap) > k+1:
                ans[j] = heap[0]
                heapq.heappop(heap)
                j += 1
            heapq.heappush(heap,arr[i])
        while len(heap):
            ans[j] = heap[0];
            heapq.heappop(heap)
            j += 1
        for i in range(n):
            arr[i] = ans[i]

File: Heaps/Sort-K-Sorted-Array/sort.cpp
#include <vector>
#include <queue>
using namespace std;

class Solution {
  public:
    void nearlySorted(vector<int>& arr, int k) {
        // code here
        int n = arr.size();
        priority_queue<int,vector<int>,greater<int>>pq;
        vector<int>ans(n,0);
        int j=0;
        for(int i=0;i<n;i++){
            if(pq.size()>k){
                ans[j] = pq.top();
                pq.pop();
                j++;
            }
            pq.push(arr[i]);
        }
        while(pq.size()){
            ans[j] = pq.top();
            pq.pop();
            j++;
        }
        for(int i=0;i<n;i++){
            arr[i] = ans[i];
        }
    }
};